modules/plugin/imagemosaic/src/test/java/org/geotools/gce/imagemosaic/ImageMosaicReaderTest.java

```diff
@@ -5174,4 +5181,45 @@ public void assertNoData(GridCoverage2D coverageDeferred, Double expectedNoData)
             assertNull(noDataContainer);
         }
     }
+
+    @Test
+    public void testHarvestWithExtraNonSpatialFile() throws Exception {
+        File source = URLs.urlToFile(rgbAURLTiff);
+        File extras = URLs.urlToFile(rgbaExtraURLTiff);
+        File testDataDir = TestData.file(this, ".");
+        File directory = new File(testDataDir, "rgba_tiff_extra_test");
```

Comment:
Naming seems strange, since it appears to only have a PNG file.

Reply:
it's a extra on top of rgba_tiff contents (adding the "png" poison pill that made the bug appear)

Reply:
Got it - its not intended to be "an extra tiff file", but "extra (broken) file to the real tiff"